tool/gravity/cli/clusterconfig.go

```diff
@@ -218,6 +221,42 @@ type configInitializer struct {
 	config   libclusterconfig.Interface
 }
 
+func validateUpdateConfig(localEnv *localenv.LocalEnvironment, config libclusterconfig.Interface) error {
+	operator, err := localEnv.SiteOperator()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cluster, err := operator.GetLocalSite()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	clusterConfig, err := operator.GetClusterConfiguration(cluster.Key())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	globalConfig := clusterConfig.GetGlobalConfig()
+	if globalConfig == nil || isCloudConfigEmpty(globalConfig) {
+		if newGlobalConfig := config.GetGlobalConfig(); newGlobalConfig != nil && !isCloudConfigEmpty(newGlobalConfig) {
+			return trace.BadParameter("cannot change cloud configuration: cluster does not have cloud provider configured")
+		}
+	}
+	if globalConfig != nil {
+		if newGlobalConfig := config.GetGlobalConfig(); newGlobalConfig != nil {
+			if newGlobalConfig.CloudProvider != "" && globalConfig.CloudProvider != newGlobalConfig.CloudProvider {
```

Comment:
Should this also check the case when `newGlobalConfig.CloudProvider` is empty but `globalConfig.CloudProvider` is not?

Reply:
This is what the if block above is about.